modules/aws/master/master.tf

```diff
@@ -7,62 +7,95 @@ locals {
 
 data "aws_availability_zones" "available" {}
 
-resource "aws_instance" "master" {
+
+resource "aws_cloudformation_stack" "master_asg" {
+  count = var.master_count
+  name  = "${var.cluster_name}-master-${count.index}"
+
+  template_body = <<EOF
+{
```

Comment:
why having CF here?

Reply:
We had the same CF for workers, I think the reason was that CF is handling the ASG update somehow better than terraform but maybe it was due to some old bug

Reply:
I will try use ASG directly

Reply:
so I tried to use ASG directly but somehow the change of ignition does not trigger ASG rolling update, no idea why.

So that is probabbly reason why we use CF which does the magic of rolling machines for us